src/lessons/Data-Structures/03.md

```diff
@@ -0,0 +1,56 @@
+<strong>decentralized web</strong>, we can all host each other's data,
+and there's a different kind of linking that's more secure, making it easy to
+trust our neighbors.
+
+## Crytographic hashing
+
+<strong>[Cryptographic hashing](https://en.wikipedia.org/wiki/Cryptographic_hash_function)</strong> is the
+most important tool in the toolbox of decentralized data structures. It opens the door to a new form of linking, known as <strong>content
+addressing</strong>, that we can use in a decentralized environment without central authorities.
+
+Hashing takes data of any size and returns you a single fixed-size "hash" for that data.
+A hash is a string of characters that looks like gobbledygook, but you can think of it
+as a link. It might look something like this:
+
+`zdpuAsHkamdCQgrDrNSwJVgjMkQWoLxdrccxV6qe9htipNein`
+
+To be honest, these content-derived links aren't currently very friendly to humans
+(`beagle.jpg` was so much more descriptive!), but they're a lot more secure. Here's why:
+
+<strong>Cryptographic hashes can be derived from the content of the data itself</strong>,
+meaning that anyone using the same
+algorithm on the same data will arrive at the same hash. If Kyle and Jess
+are both using the same decentralized web protocol, such as IPFS, to share the exact
+same photo of a kitten, both images will have exactly the same hash, or link. By comparing
+those links and confirming they're identical, we can guarantee that every single pixel
+of those two photos is identical.
+
+<strong>Cryptographic hashes are also unique</strong>. That means that if Jess uses Photoshop to remove a
+single whisker from that kitty, the udpated image will have a new hash, and it will
+be easy to tell &mdash; just from the link, without access to the file itself &mdash; that it
+contains different data.
+
+## Trust on the decentralized web
+
+On the centralized web, we've learned to trust certain authorities and not others. We do our best
+with the clues we have from URLs, but there are some malicious actors who use the shortcomings of location
+addressing to trick us.
+
+On the decentralized web, where we're all able to pitch in and host each other's data, we may not
+know much about the <em>peers</em> who are hosting data, but content addressing gives us a way to trust
+the data that they're sharing. With content addressing, we can't be fooled by malicious actors into
+accessing the wrong data. That's what makes cryptographic hashing so important to the decentralized web.
+
+## Asking peers for content
+
+With traditional location addressing, we knew we needed to visit the domain `puppies.com` to find the
+content stored as `beagle.jpg`. If the `puppies.com` domain were broken for some reason, we'd lose
+access to that image.
+
+The decentralized web works differently. When we want a specific photo of an adorable pet, we ask for it
+by its content address (hash). Who do we ask? The whole network! If Kyle is online, we'll see that he has
```

Comment:
At this point I completely had forgotten about Jess and Kyle and had to scroll up to remember.